app/src/main/java/space/taran/arknavigator/mvp/model/repo/index/PlainResourcesIndex.kt

```diff
@@ -16,142 +20,80 @@ import space.taran.arknavigator.mvp.model.repo.preview.PreviewAndThumbnail
 import space.taran.arknavigator.utils.LogTags.PREVIEWS
 import space.taran.arknavigator.utils.LogTags.RESOURCES_INDEX
 import space.taran.arknavigator.utils.listChildren
-import space.taran.arknavigator.utils.withContextAndLock
-import java.nio.file.Files
-import java.nio.file.Path
-import java.nio.file.Paths
-import kotlin.io.path.ExperimentalPathApi
-import kotlin.system.measureTimeMillis
-
-internal data class Difference(
-    val deleted: List<Path>,
-    val updated: List<Path>,
-    val added: List<Path>
-)
 
 @OptIn(ExperimentalPathApi::class)
 // The index must read from the DAO only during application startup,
 // since DB doesn't change from outside. But we must persist all changes
 // during application lifecycle into the DAO for the case of any unexpected exit.
-class PlainResourcesIndex internal constructor(
+class PlainResourcesIndex
+internal constructor(
     private val root: Path,
     private val dao: ResourceDao,
     resources: Map<Path, ResourceMeta>
 ) : ResourcesIndex {
 
-    private val mutex = Mutex()
```

Comment:
Isn't this mutex really needed?

Reply:
There have been a mutex underlying the call there have been a mutex, so it's no need.